src/Microsoft.OData.Core/ODataMessageWriterSettings.cs

```diff
@@ -189,6 +190,11 @@ public ODataUri ODataUri
         /// </summary>
         public string MultipartNewLine { get; set; }
 
+        /// <summary>
+        /// When set, type annotations will be added for derived types, even when the metadata level is set to "None".
+        /// </summary>
+        public bool AlwaysAddTypeAnnotationsForDerivedTypes { get; set; }

```

Comment:
>AlwaysAddTypeAnnotationsForDerivedTypes { get; set; } [](start = 20, length = 53)

I am ok now, however, for long term, it's better to add the "Filter" mechanism to filter the control info in the payload

Reply:
Yes -- giving consumers the ability to precisely filter the serialization of control data (within the limits set by the OData spec) seems like a good goal but I think it is outside the scope of this change.